README.md

```diff
@@ -3,37 +3,36 @@
 ![pdk-validate](https://github.com/ncsa/puppet-profile_session_log/workflows/pdk-validate/badge.svg)
 ![yamllint](https://github.com/ncsa/puppet-profile_session_log/workflows/yamllint/badge.svg)
 
-NCSA Common Puppet Profiles - Configure tlog session logging
+NCSA Common Puppet Profiles - Configure terminal session logging (tlog)
 
 ## Description
 
 Installs and configures tlog for terminal session recording. tlog data is forwarded to a remote rsyslog server.
 
-## Dependencies
-* sssd configured
-* rsyslog
-* TODO flesh this out
-
 ## Usage
 
-The goal is that no paramters are required to be set. The default paramters should work for most NCSA deployments out of the box.
+This module creates a new rsyslog ruleset named `terminal_session_log`. In order to make use of this module you need to setup rsyslog so that it calls the `terminal_session_log` ruleset (typically you want to call it as the very first action in your rules). Our existing rsyslog module doesn't have the ability to inject rules from another module, so if you are configuring rsyslog via [ncsa/profile_rsyslog](https://github.com/ncsa/puppet-profile_rsyslog) you need to make some edits to your hiera.
 
-But there are two sets of parameters you may desire to override.
+First you need to override your config for `profile_rsyslog::config_rulesets::localhost_messages:` so that you call the new rule. Putting the `terminal_session_log` ruleset at the top would look like this:
 
-# TODO ^^^^^
+```yaml
+profile_rsyslog::config_rulesets:
+  localhost_messages:
+    rules:
+      - call: 'terminal_session_log'
+#... rest of rules omitted ...
+```
 
-## Reference
+If you are using `profile_session_log::rsyslog::forward_protocol: "relp-tls"` you need to make one additional change, which is to override `profile_rsyslog::config_modules:`, specifically the config when loading the `omrelp` module:
```

Comment:
Can you document why this is needed?

I suspect it adds encryption of the log traffic data.

Reply:
Yeah, README updated with more details.